File: microservers/billetterieservice/src/main/java/io/multi/billetterieservice/domain/Region.java
package io.multi.billetterieservice.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Region {

    private Long regionId;
    private String regionUuid;
    private String libelle;
    private String code;
    private Boolean actif;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}

File: frontend/ultima-ng-19.0.0/src/app/interface/region.ts
export interface IRegion {
    regionId: number;
    regionUuid: string;
    libelle: string;
    code: string;
    actif: boolean;
    createdAt: Date | string;
    updatedAt: Date | string;
}

export interface IRegionCreateRequest {
    libelle: string;
    code?: string;
}

export interface IRegionUpdateRequest {
    libelle: string;
    code?: string;
}

export interface IRegionStatusRequest {
    actif: boolean;
}
